src/game/game.js

```diff
@@ -422,8 +422,7 @@
                 get() {
 
                     try {
-                        runCodeCache[userId].memory._parsed = JSON.parse(runCodeCache[userId].memory.get() || "{}");
-                        runCodeCache[userId].memory._parsed.__proto__ = null;
+                        runCodeCache[userId].memory._parsed = _sandboxedEval('JSON.parse')(runCodeCache[userId].memory.get() || "{}");
```

Comment:
If I'm understanding you correctly, you're saying that this will throw sometimes? 
```
exports.loop = function() {
if (Memory.__proto__ !== {}._proto__) throw "leak!";
}
```
Because I disagree.

However, I just realized something important. This means that users can overwrite JSON.parse with something of their own design to parse Memory. That's probably not what you want.